ctapipe/core/tool.py

```diff
@@ -357,6 +358,8 @@ def run(self, argv=None):
             Provenance().finish_activity(activity_name=self.name, status="interrupted")
             exit_status = 130  # Script terminated by Control-C
         except Exception as err:
+            if raises:
```

Comment:
Do we want this to happen before finishing the provenance log?
I realize that this is mostly meant for testing anyway, but would personally still write out the logs

Reply:
Good point.

Reply:
Actually, the provenance is being written (it is in the `finallly` block below, which always gets run).

But it is missing now the cause for the finishing of the activity. So we should definitely move the raise below the `finish_activity`.

Reply:
Done